src/FontMetrics.php

```diff
@@ -326,6 +339,133 @@ public function getTextWidth(string $text, $font, float $size, float $wordSpacin
         return $width;
     }
 
+    /**
+     * Returns if given style can render the text
+     *
+     * @param string $font
+     * @param string $text
+     * @return bool
+     */
+    public function canStyleRender(string $font, string $text): bool
+    {
+        if (!key_exists($font, $this->unicodeCharMapTables)) {
+            $this->loadunicodeCharMapTables($font);
+        }
+
+        return $this->intersectUnicodeAgainstFont($text, $font);
+    }
+
+    private function intersectUnicodeAgainstFont(string $text, string $font): bool
+    {
+        if (!key_exists($font, $this->unicodeCharMapTables)) {
+            return false;
+        }
+
+        $range = $this->getCanvas()->get_cpdf()->utf8toCodePointsArray($text);
+        $glyphIndexArray = $this->unicodeCharMapTables[$font]["glyphIndexArray"];
+        unset($glyphIndexArray[0xFFFF]);
+
+        return array_intersect($range, array_keys($glyphIndexArray)) === $range;
+    }
+
+    /**
+     * Tries names for supported fonts to easily find font file
+     * @param string $fontFamily
+     * @return string
+     */
+    public function tryFiles(string $fontFamily): string
+    {
+        $possibleFontExtensions = ['ttf', 'otf', 'woff', 'woff2'];
+        foreach ($possibleFontExtensions as $extension) {
+            $fontFile = sprintf('%s.%s', $fontFamily, $extension);
+            if (file_exists($fontFile)) {
+                return $fontFile;
+            }
+        }
+
+        return $fontFamily;
+    }
+
+    public function loadunicodeCharMapTables($font): array
+    {
+        if (key_exists($font, $this->unicodeCharMapTables)) {
+            return $this->unicodeCharMapTables[$font];
+        }
+
+        $fontFile = $this->tryFiles($font);
+        if (!file_exists($fontFile)) {
+            return [];
+        }
+
+        $fontObj = Font::load($fontFile);
+        $fontObj->parse();
+
+        // Find Unicode char map table
+        $subtable = null;
+        foreach ($fontObj->getData("cmap", "subtables") as $_subtable) {
+            if ($_subtable["platformID"] == 0 || $_subtable["platformID"] == 3 && $_subtable["platformSpecificID"] == 1) {
+                $subtable = $_subtable;
+                break;
+            }
+        }
+
+        $this->unicodeCharMapTables[$font] = $subtable;
+        return $this->unicodeCharMapTables[$font];
+    }
+
```

Comment:
Add the following method in support of the changes to getTextFamilies:

```suggestion

    private function fontSupportsGlyph(int $codepoint, string $font): bool
    {
        if (!key_exists($font, $this->unicodeCharMapTables) || $codepoint === 0xFFFF) {
            return false;
        }

        $fontRange = $this->unicodeCharMapTables[$font];
        $glyphIndexArray = $fontRange["glyphIndexArray"];

        return array_key_exists($codepoint, $glyphIndexArray);
    }

```